tests/Console/ConfigurationResolverTest.php

```diff
@@ -961,6 +961,26 @@ public function provideResolveRulesCases()
                     '@A' => true,
                 ),
             ),
+            '@A - @C' => array(
```

Comment:
**It is** more complicated. :no_mouth: So I'm dropping in kinda awkwardly in hope to help however I can.

Okay, so I partially agree with @julienfalque on `@Foo - @Bar` should read as `remove from @Foo all rules that are present in @Bar`. E.g.: considering:

`@Foo => { 'A' => true, 'B' => true, 'C' => true}`
`@Bar => { 'A' => true, 'C' => false}`
then `{@Foo - @Bar} => {'B' => true, 'C' => true}`

In cases of subtraction, the most coherent, imho, would be to ignore rules set to false within the subrahend. Therefore the `'C' => false` being ignored. I imagine someone doing something like `@Symfony - @PSR2` and not expecting to have some rule turned off in `@PSR2` being automatically turned on.

So it would be like mathematical union, intersection and subtraction between groups: 
![shaded-part-represents-b-a](https://cloud.githubusercontent.com/assets/14227063/24971166/5185b8b2-1f8d-11e7-9e9b-7ae02908bd44.png)

I believe the effect of setting a group of rules as `false` should be `disable all rules within the group`. So, in cases of sequence definition, I'd go with:

`@Foo => { 'A' => true, 'B' => true, 'C' => true}`
`@Bar => { 'A' => true, 'C' => false}`
then `{@Foo = true,  @Bar = false} => {'B' => true}`

That being said, isn't the subtraction between groups of rules a bit of an edge case? It could be accomplished setting the rules independently. Seems like a bit of over engineering. Or maybe someone could provide some use cases?

What do you guys think?